lib/mutant/diff.rb

```diff
@@ -74,19 +72,15 @@ def hunks
       end
     end
 
-    # Minimized hunks
+    # Minimized hunk
     #
-    # @return [Array<Diff::LCS::Hunk>]
-    def minimized_hunks
+    # @return Diff::LCS::Hunk
+    def minimized_hunk
       head, *tail = hunks
 
-      tail.each_with_object([head]) do |right, aggregate|
-        left = aggregate.last
-        if right.overlaps?(left)
-          right.merge(left)
-          aggregate.pop
-        end
-        aggregate << right
+      tail.reduce(head) do |left, right|
+        right.merge(left)
+        right
```

Comment:
Proactive comment: Yes `Diff::LCS::Hunk#merge` returns `nil` but modifies receiver in-place.